lightkurve/correctors/cbvcorrector.py

```diff
@@ -1,258 +1,946 @@
-            # where p1 is the posterior probability (evaluated at the MAP)
-            # for the model with n-2 cbvs and p2 is the posterior probability
-            # also evaluated at the MAP for the model with n-1 cbvs
-        k = np.argmin(self.bayes_factor)
-        # transform to get the actual Bayes factor
-        self.bayes_factor = np.exp(-np.array(self.bayes_factor))
-        # the k+2 here comes from the fact that Python indexes begin
-        # from 0 and we count CBVs starting from 1 and also
-        # note that range(1, k) equals the interval [1, k), which excludes k.
-        return list(range(1, k+2))
-
-    def get_cbv_url(self):
-        # gets the html page and finds all references to 'a' tag
-        # keeps the ones for which 'href' ends with 'fits'
-        # this might slow things down in case the user wants to fit 1e3 stars
-        soup = BeautifulSoup(requests.get(self.cbv_base_url).text, 'html.parser')
-        cbv_files = [fn['href'] for fn in soup.find_all('a') if fn['href'].endswith('fits')]
+        assert module > 1 and module < 85, 'Invalid module number'
+        assert output > 0 and output < 5, 'Invalid output number'
+
+        # Get the mission: Kepler or K2
+        # Sadly, the HDU does not explicitly say if this is Kepler or K2 CBVs.
+        if HDU['PRIMARY'].header.__contains__('QUARTER'):
+            mission = 'Kepler'
+        elif HDU['PRIMARY'].header.__contains__('CAMPAIGN'):
+            mission = 'K2'
+        else:
+            raise Exception('This does not appear to be a Kepler or K2 FITS HDU')
+
+
+        extName = 'MODOUT_{0}_{1}'.format(module, output)
+
+        try:
+            # Read the columns and meta data
+            dataTbl = Table.read(HDU[extName], format="fits")
+            dataTbl.meta.update(HDU[0].header)
+            dataTbl.meta.update(HDU[extName].header)
+  
+            # TimeSeries-based objects require a dedicated time column
+            # Replace NaNs with default time '2000-01-01', otherwise,
+            # astropy.time.Time complains
+            nanHere = np.nonzero(np.isnan(dataTbl['TIME_MJD'].data))[0]
+            timeData = dataTbl['TIME_MJD'].data
+            timeData[nanHere] = Time(['2000-01-01'], scale='utc').mjd
+            cbvTime = Time(timeData, format='mjd')
+            dataTbl.remove_column('TIME_MJD')
+            
+            # Gaps are labelled as 'GAPFLAG' so rename!
+            dataTbl['GAP'] = dataTbl['GAPFLAG']
+            dataTbl.remove_column('GAPFLAG')
+
+            dataTbl.meta['MISSION'] = mission
+            dataTbl.meta['CBV_TYPE'] = 'SingleScale'
+            
+        except:
+            dataTbl = None
+            cbvTime = None
+
+        # Here we instantiate the actual object
+        return self(data=dataTbl, time=cbvTime, **kwargs)
 
-        if self.lc.mission == 'Kepler':
-            quarter = 'q{:02}'.format(self.lc.quarter)
+
+    @property
+    def mission(self):
+        return self.meta['MISSION']
```

Comment:
I recommend using `meta.get(key, default=None)`, which will return `default` if `key` is missing from the dictionary:
```suggestion
        return self.meta.get('MISSION')
```

Reply:
changed.